src/doc/man/cargo-bench.md

```diff
@@ -5,7 +5,7 @@
 
 ## NAME
 
-cargo-bench - Execute benchmarks of a package
+cargo-bench --- Execute benchmarks of a package
```

Comment:
You need to regenerate manpages by running `./src/doc/build-man.sh`. However, [`mdman`](https://github.com/rust-lang/cargo/tree/master/crates/mdman) doesn't take smart punctuations at all. Three dashes `---` will be displayed verbatim when you run `cargo help build`. I feel like it is a regression.

There are a few options here:

* Make `mdman` convert punctuations as the same as what `mdbook` have done. You need at least upgrade `pulldown-cmark` for `mdman` to [v0.8.0](https://github.com/raphlinus/pulldown-cmark/releases/tag/v0.8.0) or newer to get that feature.
* Don't use smart punctuations for pages that are shared between manpages and The Cargo Book. Instead, manually check and fix them. This diverges our docs though.
* Completely drop the idea of using smart punctuations. Check and fix all punctuations manually for all docs. Not really automatic, yet if we fix it this time, we have less burden for the next punctuation cleanup.

I am not sure which is the best.

Reply:
BTW, here is the place if you go the route to enabling smart punctuations.

https://github.com/rust-lang/cargo/blob/e54a0b5f11d025c59df003773d927439a8778eaf/crates/mdman/src/lib.rs#L67-L71

Reply:
I'll try enabling it.

Reply:
Please help me out. I'm not familiar with .1 files.

The conversion command for .1 files gave me error messages about special characters, originating from the following function. See my comments at the starts of lines:

```rust
fn escape(s: &str) -> Result<String, Error> {
    /*ADDED BY ME*/ // Note: Possible source on output escape sequences: https://man7.org/linux/man-pages/man7/groff_char.7.html

    let mut replaced = s
        .replace('\\', "\\(rs")
        .replace('-', "\\-")
        .replace('\u{00A0}', "\\ ") // non-breaking space (non-stretchable)
        .replace('–', "\\[en]") // \u{2013} en-dash
        .replace('—', "\\[em]") // \u{2014} em-dash
        /*ADDED BY ME*/ .replace('‘', "\\[oq]") // \u{2018} left single quote
        /*ADDED BY ME*/ .replace('’', "\\[cq]") // \u{2019} right single quote or apostrophe
        /*ADDED BY ME*/ .replace('“', "\\[lq]") // \u{201c} left double quote
        /*ADDED BY ME*/ .replace('”', "\\[rq]") // \u{201d} right double quote
        /*NOT SURE ABOUT EL. WHAT'S THE DEFINITIVE SOURCE ON THIS?*/ .replace('…', "\\[el]") // \u{2026} ellipsis
        .replace('│', "|") // \u{2502} box drawing light vertical (could use \[br])
        .replace('├', "|") // \u{251C} box drawings light vertical and right
        .replace('└', "`") // \u{2514} box drawings light up and right
        .replace('─', "\\-") // \u{2500} box drawing light horizontal
    ;
    if replaced.starts_with('.') {
        replaced = format!("\\&.{}", &replaced[1..]);
    } else if replaced.starts_with('\'') {
        /*WHAT TO DO ABOUT THIS? CURLY VERSION HANDLED ABOVE*/ replaced = format!("\\(aq{}", &replaced[1..]);
    }

    if let Some(ch) = replaced.chars().find(|ch| {
        !matches!(ch, '\n' | ' ' | '!'..='/' | '0'..='9'
            | ':'..='@' | 'A'..='Z' | '['..='`' | 'a'..='z' | '{'..='~')
    }) {
        bail!(
            "character {:?} is not allowed (update the translation table if needed)",
            ch
        );
    }
    Ok(replaced)
}
```

BTW, why does `escape()` use `\(xx` at all when `\[xx]` seems to be equivalent?

Reply:
The .md files generated in `src/doc/src/commands` get `'` changed to `’`, and `...` to `…`, but `---` stays `---`. Is that okay, or a bug somewhere?